### PR TITLE
Add covid symptom annotator

### LINKED ISSUE #192: COVID-19 Sign/symptom extractor
### Tool description

Extracting COVID-19 signs and symptoms per the CDC guideline from clinical narrative texts.

### Tool input

Raw COVID-19 related texts

### Tool output

Annotations of signs and symptoms, including span, category, and modifiers such as if the condition is negated or current.  

### Tool example

OHNLP MedTagger: https://github.com/OHNLP/MedTagger 

### Dataset

Yes

### Contact information

**Your name:** Sijia Liu
**Your email address:** liu.sijia@mayo.edu
**Name of your organization:** Mayo Clinic

### PR DESCRIPTION
**Pull request by tschaffter**
Fixes #192 

## Task

- [x] @tschaffter to suggest the initial schemas
- [x] @ymnliu to complete the PR
- [x] @tschaffter and selected team members should approve the PR (@yy6linda, @cascadianblue, @gkowalski)

### CODE REVIEW

**Comment by tschaffter** [comment]
@ymnliu Should the annotator be called `covid19-symptom-annotator`? It's more descriptive, but if we expect this annotator to work with other covid-x, we may prefer to keep it's name more generic..

**Comment by ymnliu** [comment]
Hi @tschaffter would you like to review this PR?

**Comment by tschaffter** [comment]
@ymnliu I fixed a conflict that was blocking this branch from starting the tests.

**Comment by tschaffter** [comment]
@ymnliu I cannot review this PR since I created it. Below are my comments:

- I should not have removed the schemas listed in your commit "remove unrelated annotations". These schemas are used by other tools. This repository lists schemas for many tools, not just the COVID annotator.
- The convention in this repo is to use snake case for enumeration (e.g. `nasal_obstruction` instead of `Nasal obstruction`).
- The documentation of the tool in the file `openapi/covid-symptom-annotator/openapi.yaml` must be updated.

**Comment by tschaffter** [comment]
@ymnliu You can use [git revert](https://www.gitkraken.com/learn/git/problems/revert-git-commit) to revert the commit where you remove the files that must be kept.

**Comment by ymnliu** [comment]
Hi @tschaffter I was removing the deid scheme files from the branch, which are included from the very beginning of the branch. I assume I should use git rm instead? Anyway, I will fix it based on your review comments.

**Comment by tschaffter** [comment]
> I was removing the deid scheme files from the branch, which are included from the very beginning of the branch.

You should not remove any files from the branch. When I created the branch used for this PR, the branch was a copy of the default branch of this repo (`develop`). Then we only need to add/update files needed to add the specification of the new COVID symptom annotation tool. When merging this PR, the changes that we made in this branch will be applied to the default branch. If you remove the file `openapi/commons/components/schemas/TextPersonNameAnnotation.yaml` as you did, this file would be removed from the default branch, which would break the specification of the Person Name Annotation tool. One of the reasons the checks are failing in this PR is because the series of tests I put in placed detect this break.

With `git revert 2561caaa27b5d151d8c22034e1413c5136894675`, you should be able to undo the commit `remove unrelated annotations` where you are remove files that we need to keep in this branch.

File: openapi/commons/components/schemas/TextCovidSymptomAnnotation.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          - dyspnea
+          - chest_pain
+          - delirium
+          - hypersomnia
+          - cyanosis
+example:
+  start: 42
+  length: 9
+  text: dry cough
+  dateFormat: MM/DD/YYYY
```
Comment (yy6linda) [inline openapi/commons/components/schemas/TextCovidSymptomAnnotation.yaml:45]: need to have a new name other than the "dateFormat", maybe "symptom_type"?

File: openapi/covid-symptom-annotator/components/schemas/TextCovidSymptomAnnotationRequest.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+type: object
+description: A request to annotate COVID symptom in a clinical note
```
Comment (yy6linda) [inline openapi/covid-symptom-annotator/components/schemas/TextCovidSymptomAnnotationRequest.yaml:2]: COVID symptoms

File: openapi/commons/components/schemas/TextCovidSymptomAnnotation.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+type: object
+description: A COVID symptom annotation in a text
+allOf:
+  - $ref: TextAnnotation.yaml
+  - type: object
+    properties:
+      certainty:
+        description: >
+          Whether the annotation is positively or negatively corelated with the
```
Comment (cascadianblue) [inline openapi/commons/components/schemas/TextCovidSymptomAnnotation.yaml:9]: `corelated` -> `correlated`

Comment (cascadianblue) [reply openapi/commons/components/schemas/TextCovidSymptomAnnotation.yaml:9]: Wouldn't this just be a function of `condition` (i.e. certain conditions are positively correlated with COVID-19, some are not)?

File: openapi/commons/components/schemas/TextCovidSymptomAnnotation.yaml (hunk 1 above)
Comment (cascadianblue) [reply openapi/commons/components/schemas/TextCovidSymptomAnnotation.yaml:45]: This line should be

```
condition: cough
```

File: openapi/covid-symptom-annotator/openapi.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    url: https://nlpsandbox.io
+  license:
+    name: Apache 2.0
+    url: https://github.com/nlpsandbox/nlpsandbox-schemas/blob/develop/LICENSE
+  x-logo:
+    url: https://nlpsandbox.github.io/nlpsandbox-schemas/logo.png
+  description: >
+    # Introduction
+
+    The COVID Symptom Annotator is one of the first type of NLP Tools that can be
```
Comment (cascadianblue) [inline openapi/covid-symptom-annotator/openapi.yaml:17]: "one of the first types of NLP tools"

File: openapi/covid-symptom-annotator/openapi.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    - The output of this tool must be reproducible: a given input should always
+      generate the same output.
+
+    - This tool must not attempt to connect to remote server for reproducibility,
+      robustness, and security reasons. When benchmarked on [nlpsandbox.io](https://nlpsandbox.io),
+      this tool will not be able to connect to remote servers.
+
+    # Examples
+
+    - [COVID Symptom Annotator Example (Python)](https://github.com/nlpsandbox/covid-symptom-annotator-example)
```
Comment (cascadianblue) [inline openapi/covid-symptom-annotator/openapi.yaml:54]: URL currently not reachable -- which might be intended

**Review by boyleconnor — CHANGES_REQUESTED** [review]

File: openapi/covid-symptom-annotator/openapi.yaml (hunk 4 above)
Comment (tschaffter) [reply openapi/covid-symptom-annotator/openapi.yaml:17]:
```suggestion
    The COVID Symptom Annotator is one of the first NLP Sandbox tools that can be
```

Comment (tschaffter) [reply openapi/covid-symptom-annotator/openapi.yaml:17]: I now use the above suggestion in all other tool specifications.

File: openapi/covid-symptom-annotator/openapi.yaml (hunk 5 above)
Comment (tschaffter) [reply openapi/covid-symptom-annotator/openapi.yaml:54]: Yes, an example implementation will be added shortly and before the next release of the NLP Sandbox schemas.

**Pull request merged**